pkg/debugapi/welcome_message_test.go

```diff
@@ -0,0 +1,66 @@
+package debugapi_test
```

Comment:
@mortelli - to reply to your question regarding the RWMutex (and the potentially unfortunate Synced naming):

My understanding is that the welcome message is read concurrently as there are concurrent connections to peers. A RWMutex would make sure that the value of the welcome message should stay synced - as in it should prevent a dirty read. The routine writing the welcome message would block until all routines doing handshakes complete the read.

I did this as an extra safety precaution, although I must admit I cannot estimate the performance implication on the handshake.

Lastly the name is kinda unfortunate - I wanted to attract attention to the fact that it could block.

Reply:
> @mortelli - to reply to your question regarding the RWMutex (and the potentially unfortunate Synced naming):
> 
> My understanding is that the welcome message is read concurrently as there are concurrent connections to peers. A RWMutex would make sure that the value of the welcome message should stay synced - as in it should prevent a dirty read. The routine writing the welcome message would block until all routines doing handshakes complete the read.
> 
> I did this as an extra safety precaution, although I must admit I cannot estimate the performance implication on the handshake.

i see. so we have a potential problem if through the API we are trying to set a new welcome message, while it's being read in other threads that are doing handshakes.

so this would be a fairly rare case, but it can definitely happen and it can certainly cause a `panic`. 
 
> Lastly the name is kinda unfortunate - I wanted to attract attention to the fact that it could block.

fair enough. i would switch it to `WelcomeMessage` or `GetWelcomeMessage` (although i think getter and setter wording is not so idiomatic in go) but i do not have a strong preference for either.

Reply:
I was never really sure about the idiomatic wording, but I agree.